src/mlpack/core/data/dictionary_encoding.hpp

```diff
@@ -0,0 +1,132 @@
+/**
+ * @file dictionary_encoding.hpp
+ * @author Jeffin Sam
+ *
+ * Implementation of dictionary encoding functions.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_DATA_DICT_ENCODING_HPP
+#define MLPACK_CORE_DATA_DICT_ENCODING_HPP
+
+#include <mlpack/prereqs.hpp>
+#include "mlpack/core/boost_backport/boost_backport_string_view.hpp"
+#include <boost/functional/hash.hpp>
+
+namespace mlpack {
+namespace data {
+
+/*
+* Class Hasher importanant to remove inablitiy of unordered_map to store
+* boost::string_view as key.
+*/
+class Hasher
```

Comment:
I think we shouldn't declare this class at the global scope. Probably it is better to move the definition to `DictionaryEncoding`. And it might be a better way to declare a specialization of `boost::hash` for some boost versions.

Reply:
Done .